reducer/src/main/java/com/graphicsfuzz/reducer/reductionopportunities/GlobalVariableDeclToExprReductionOpportunities.java

```diff
@@ -17,18 +17,56 @@
 package com.graphicsfuzz.reducer.reductionopportunities;
 
 import com.graphicsfuzz.common.ast.TranslationUnit;
+import com.graphicsfuzz.common.ast.decl.FunctionDefinition;
+import com.graphicsfuzz.common.ast.decl.ScalarInitializer;
+import com.graphicsfuzz.common.ast.decl.VariableDeclInfo;
+import com.graphicsfuzz.common.ast.decl.VariablesDeclaration;
+import com.graphicsfuzz.common.ast.type.TypeQualifier;
 import com.graphicsfuzz.common.transformreduce.ShaderJob;
 import com.graphicsfuzz.common.util.ListConcat;
+import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
 
+/*
+ * This class finds opportunities to remove initializers from global variable declarations and
+ * replace them with assignment statements which will be inserted as the first statement in main
+ * function. If main function is not found, do not consider finding this opportunities.
+ * For example, in:
+ * int a = 1;
+ * int b = foo();
+ * void main(){
+ *  int c = 1;
+ * }
+ *
+ * <p>Because each of a and b has an initializer, we can transform the code fragment
+ * into the following:
+ * int a;
+ * int b;
+ * void main(){
+ *  a = 1;
+ *  b = foo();
+ *  int c = 1;
+ * }
+ */
+
 public class GlobalVariableDeclToExprReductionOpportunities
     extends ReductionOpportunitiesBase<GlobalVariableDeclToExprReductionOpportunity> {
+  private final List<VariablesDeclaration> globalVariableDecl;
+
   public GlobalVariableDeclToExprReductionOpportunities(TranslationUnit tu,
                                                         ReducerContext context) {
     super(tu, context);
```

Comment:
I favour always using "this." in a constructor, just for uniformity.

Reply:
Fixed